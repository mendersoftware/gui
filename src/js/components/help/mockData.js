// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
