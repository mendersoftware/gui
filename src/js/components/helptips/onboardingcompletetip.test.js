// Copyright 2019 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import React from 'react';
import { Provider } from 'react-redux';

import { act, waitFor } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import OnboardingCompleteTip from './onboardingcompletetip';

const mockStore = configureStore([thunk]);

describe('OnboardingCompleteTip Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      ...defaultState,
      app: {
        ...defaultState.app,
        features: {
          ...defaultState.app.features,
          hasMultitenancy: true,
          isHosted: true
        }
      }
    });
    jest.spyOn(global, 'encodeURIComponent').mockImplementationOnce(() => 'http%3A%2F%2Ftest.com');
  });

  it('renders correctly', async () => {
    const ui = (
      <Provider store={store}>
        <OnboardingCompleteTip targetUrl="https://test.com" />
      </Provider>
    );
    const { baseElement, rerender } = render(ui);
    await act(async () => {});
    await waitFor(() => rerender(ui));
    const view = baseElement;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
