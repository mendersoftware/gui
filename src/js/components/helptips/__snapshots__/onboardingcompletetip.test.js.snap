// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingCompleteTip Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.3929rem;
}

.emotion-0 iconButton {
  margin-right: 8px;
}

.emotion-1 {
  z-index: 1500;
  pointer-events: auto;
  opacity: 0.9;
}

.emotion-2 {
  background-color: rgba(188, 188, 188, 0.92);
  border-radius: 4px;
  color: #fff;
  font-family: Lato,sans-serif;
  padding: 4px 8px;
  font-size: 0.6384rem;
  max-width: 300px;
  margin: 2px;
  word-wrap: break-word;
  font-weight: 500;
  background-color: rgb(50, 50, 50);
  background-color: #337a87;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  color: #e9e9e9;
  font-size: 14px;
  max-width: 350px;
  padding: 12px 18px;
  width: 350px;
}

.MuiTooltip-popper[data-popper-placement*="left"] .emotion-2 {
  transform-origin: right center;
  margin-right: 14px;
}

.MuiTooltip-popper[data-popper-placement*="right"] .emotion-2 {
  transform-origin: left center;
  margin-left: 14px;
}

.MuiTooltip-popper[data-popper-placement*="top"] .emotion-2 {
  transform-origin: center bottom;
  margin-bottom: 14px;
}

.MuiTooltip-popper[data-popper-placement*="bottom"] .emotion-2 {
  transform-origin: center top;
  margin-top: 14px;
}

.emotion-2 a {
  color: #e9e9e9;
}

.emotion-2.MuiTooltip-tooltipPlacementTop {
  margin-left: 30px;
  margin-bottom: 0;
  margin-top: calc(30 + 12px);
}

.emotion-2.MuiTooltip-tooltipPlacementRight {
  margin-top: 15px;
}

.emotion-2.MuiTooltip-tooltipPlacementBottom {
  margin-left: 30px;
}

.emotion-2.MuiTooltip-tooltipPlacementLeft {
  margin-top: 15px;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  background-color: #fff;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #337a87;
  border-radius: 2px;
  font-size: 14px;
  font-weight: bold;
  padding: 10px 15px;
  background-color: #fff;
  background-color: #fff;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  background-color: #fff;
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(51, 122, 135, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-3:hover {
  colors: #337a87;
}

.emotion-3.MuiButton-text {
  color: rgba(10, 10, 11, 0.78);
}

.emotion-3:hover {
  background-color: #fff;
}

.emotion-3:hover {
  background-color: #fff;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #5d0f43;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  border-radius: 2px;
  font-size: 14px;
  font-weight: bold;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(65, 10, 46);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: #5d0f43;
  }
}

.emotion-5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-5:hover {
  colors: #337a87;
}

.emotion-5.MuiButton-text {
  color: rgba(10, 10, 11, 0.78);
}

<body>
  <div>
    <div
      aria-labelledby="onboarding-finished"
      class="tooltip onboard-icon onboard-tip"
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
  </div>
  <div
    class="MuiTooltip-popper MuiTooltip-popperInteractive emotion-1 MuiPopper-root"
    data-popper-escaped=""
    data-popper-placement="bottom"
    data-popper-reference-hidden=""
    id="onboarding-finished"
    role="tooltip"
    style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom emotion-2"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="content"
      >
        <p>
          Great work! You updated your device with the new Release!
        </p>
        <div
          class="margin-bottom-small margin-top-small"
        >
          Your device is now running the updated version of the software. At
          <div
            class="flexbox centered"
            style="margin: 5px 0px;"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button emotion-3"
              href="http://192.168.10.141:85/index.html?source=http%3A%2F%2Ftest.com"
              tabindex="0"
              target="_blank"
            >
              Go to http://192.168.10.141:85
              <span
                class="MuiTouchRipple-root emotion-4"
              />
            </a>
          </div>
          you should now see "Hello world" in place of the webpage you saw previously. If you continue to see the webpage you saw previously you might have to refresh the page.
        </div>
        <p>
          You've now got a good foundation in how to use Mender. Look for more help hints in the UI as you go along.
        </p>
        What next?
        <div>
          Proceed to one of the following tutorials (listed in recommended order):
          <ol>
            <li>
              <a
                href="https://docs.mender.io/get-started/deploy-an-operating-system-update"
                rel="noopener noreferrer"
                target="_blank"
              >
                Deploy an operating system update
              </a>
            </li>
            <li>
              <a
                href="https://docs.mender.io/get-started/deploy-a-container-update"
                rel="noopener noreferrer"
                target="_blank"
              >
                Deploy a container update
              </a>
            </li>
          </ol>
        </div>
        <div
          class="flexbox"
        >
          <div
            style="flex-grow: 1;"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium emotion-5"
            tabindex="0"
            type="button"
          >
            Close
            <span
              class="MuiTouchRipple-root emotion-4"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
